plugins/errorlogs/frontend/public/javascripts/countly.models.js

```diff
@@ -11,16 +12,42 @@
             url: countlyCommon.API_PARTS.data.r + "/errorlogs",
             data: {
                 "app_id": countlyCommon.ACTIVE_APP_ID,
-                "bytes": 100000
+                "bytes": 1
             },
             success: function(json) {
-                _data = json;
+                _list = [];
+                for (var k in json) {
+                    _list.push({name: k + " Log", value: k});
+                }
             }
         });
     };
 
-    countlyErrorLogs.getData = function() {
-        return _data;
+    countlyErrorLogs.getLogNameList = function() {
+        return _list;
+    };
+
+    countlyErrorLogs.getLogCached = function() {
+        return _logCache;
+    };
+
+    countlyErrorLogs.getLogByName = function(logName, callback) {
+        return $.ajax({
+            type: "GET",
+            url: countlyCommon.API_PARTS.data.r + "/errorlogs",
+            data: {
+                "app_id": countlyCommon.ACTIVE_APP_ID,
+                "bytes": 100000,
+                "log": logName
+            },
+            success: function(data) {
+                _logCache = {};
+                _logCache[logName] = data;
+            },
+            complete: function() {
+                callback && callback();
```

Comment:
Codacy found an issue: [Expected an assignment or function call and instead saw an expression.](https://app.codacy.com/manual/ar2rsawseen/countly-server/pullRequest?prid=5953117)